contracts/TalentLayerService.sol

```diff
@@ -263,9 +272,10 @@ contract TalentLayerService is Initializable, ERC2771RecipientUpgradeable, UUPSU
         string calldata _dataUri,
         uint256 _expirationDate
     ) public payable onlyOwnerOrDelegate(_profileId) {
-        require(allowedTokenList[_rateToken], "This token is not allowed");
+        require(allowedTokenList[_rateToken].isWhitelisted, "This token is not allowed");
         uint256 proposalPostingFee = talentLayerPlatformIdContract.getProposalPostingFee(_platformId);
         require(msg.value == proposalPostingFee, "Non-matching funds");
+        require(allowedTokenList[_rateToken].minimumTransactionAmount <= _rateAmount, "Amount is too low");
```

Comment:
require(_rateAmount >= allowedTokenList[_rateToken].minimumTransactionAmount, "Amount is too low");